### PR TITLE
change: use mpi4py in MPI command for Python executables

### PR DESCRIPTION
**Pull request by laurenyu**
**do not merge** until https://github.com/aws/sagemaker-tensorflow-container/pull/185 is released

MPI may fail to detect process failure, which can cause a process to hang. `mpi4py` forces all processes to abort if an uncaught exception occurs. See https://docs.chainer.org/en/stable/chainermn/tutorial/tips_faqs.html#mpi-process-hangs-after-an-unhandled-python-exception (though the link is related to Chainer, it says that the problem is not specific to ChainerNM).

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

- [x] I have read the [CONTRIBUTING](https://github.com/aws/sagemaker-containers/blob/master/CONTRIBUTING.md) doc
- [x] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/sagemaker-containers/blob/master/CONTRIBUTING.md#commit-message-guidlines)
- [x] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [ ] I have updated any necessary [documentation](https://github.com/aws/sagemaker-containers/blob/master/README.rst) (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by jesterhazy** [comment]
### AWS CodeBuild CI Report

* Result: SUCCEEDED
* [Build Logs](https://5jg9tp8agh.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=b5909db7-9d60-41f8-b1ad-e6ef85373942%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

File: src/sagemaker_containers/_mpi.py [hunk 1]
```diff
@@ -174,6 +174,11 @@ def _create_command(self):  # type: () -> List[str, Any]
 
         return command
 
+    def _python_executable(self):
```
Comment (mvsusp) [inline src/sagemaker_containers/_mpi.py]: Maybe the name ```_python_executable``` implies that this function returns the location of the Python executable? I honestly don't have a better name suggestion though.

Comment (laurenyu) [reply src/sagemaker_containers/_mpi.py]: that's a fair point. how about `_python_command`?

Comment (mvsusp) [reply src/sagemaker_containers/_mpi.py]: sounds good!

**Comment by jesterhazy** [comment]
### AWS CodeBuild CI Report

* Result: SUCCEEDED
* [Build Logs](https://5jg9tp8agh.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=f97f7ac3-1802-4d29-8dd7-980b402a1f98%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Pull request merged**